crates/sui-rpc-benchmark/src/json_rpc/runner.rs

```diff
@@ -54,37 +84,205 @@ impl JsonRpcStats {
     }
 }
 
+impl PaginationCursorState {
+    fn with_window_size(window: Duration) -> Self {
+        Self {
+            requests: HashMap::new(),
+            window,
+        }
+    }
+
+    /// Returns the index of the cursor parameter for a method, if it exists;
+    /// Otherwise, it means no cursor transformation is needed for this method.
+    fn get_method_cursor_index(method: &str) -> Option<usize> {
+        METHOD_CURSOR_POSITIONS.get(method).copied()
+    }
+
+    fn get_method_key(method: &str, params: &[Value]) -> Result<String, anyhow::Error> {
+        let cursor_idx = METHOD_CURSOR_POSITIONS
+            .get(method)
+            .ok_or_else(|| anyhow::anyhow!("method {} not found in cursor positions", method))?;
+        let key_param_len = params.len();
+        let mut key_params = params.to_vec();
+        let param_to_modify = key_params.get_mut(*cursor_idx).ok_or_else(|| {
+            anyhow::anyhow!(
+                "params length {} is less than cursor index {} for method {}",
+                key_param_len,
+                cursor_idx,
+                method
+            )
+        })?;
+        *param_to_modify = Value::Null;
+        serde_json::to_string(&key_params)
+            .map(|params_str| format!("{}-{}", method, params_str))
+            .map_err(|e| anyhow::anyhow!("failed to generate key for method {}: {}", method, e))
+    }
+
+    fn update_params_cursor(
+        params: &mut Value,
+        cursor_idx: usize,
+        new_cursor: Option<&Value>,
+        method: &str,
+    ) -> Result<(), anyhow::Error> {
+        let params_array = params
+            .get_mut("params")
+            .and_then(|v| v.as_array_mut())
+            .ok_or_else(|| {
+                anyhow::anyhow!("params not found or not an array for method {}", method)
+            })?;
+        if params_array.len() <= cursor_idx {
+            return Err(anyhow::anyhow!(
+                "cursor index {} is out of bounds for method {}",
+                cursor_idx,
+                method
+            ));
+        }
```

Comment:
As per the earlier comment, we should be able to handle this case.

```suggestion
        if params_array_len() <= cursor_idx {
            params_array.resize(cursor_idx + 1, json!(null));
        }
```